libstuff/SSignal.cpp

```diff
@@ -173,31 +182,57 @@ void _SSignal_StackTrace(int signum, siginfo_t *info, void *ucontext) {
         if (!_SSignal_threadCaughtSignalNumber) {
             _SSignal_threadCaughtSignalNumber = signum;
 
+            SWARN("Signal " << strsignal(_SSignal_threadCaughtSignalNumber) << "(" << _SSignal_threadCaughtSignalNumber << ") caused crash, logging stack trace.");
+
             // What we'd like to do here is log a stack trace to syslog. Unfortunately, neither computing the stack
             // trace nor logging to to syslog are signal safe, so we try a couple things, doing as little as possible,
             // and hope that they work (they usually do, though it's not guaranteed).
 
             // Build the callstack. Not signal-safe, so hopefully it works.
-            void* callstack[100];
-            int depth = backtrace(callstack, 100);
-
-            // Log it to a file. Everything in this block should be signal-safe, if we managed to generate the
-            // backtrace in the first place.
-            int fd = creat("/tmp/bedrock_crash.log", 0666);
-            if (fd != -1) {
-                backtrace_symbols_fd(callstack, depth, fd);
-                close(fd);
+            void** callstack{0};
+            int max_depth = 10;
+            int depth{0};
+            while (true) {
+                if (callstack) {
+                    free(callstack);
+                }
+                callstack = (void**)malloc(sizeof(void*) * max_depth);
+                depth = backtrace(callstack, max_depth);
```

Comment:
`backtrace` here is a gnu/linux command, right https://www.gnu.org/software/libc/manual/html_node/Backtraces.html?